### PR TITLE
Feat/migrate deposit flow to easy swapper v2

### PR DESCRIPTION
**Pull request by dimlbc**
TODO: 

- [x] add missing easy swapper addresses on Optimism/Arbitrum/Polygon

### CODE REVIEW

**Comment by D-Ig** [comment]
[updated abi here](https://github.com/dhedge/dhedge-v2/blob/feat/swapper-v2/abi/EasySwapperV2.json)

File: packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-deposit-lock-time.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const { chainId, address } = useTradingPanelPoolConfig()
+  const { data: { customCooldown } = {} } = usePoolStatic({
+    address,
+    chainId,
+  })
+  const { depositMethod } = useVaultDepositParams()
+
+  const customLockTime = customCooldown
+    ? formatDuration({ minutes: Number(customCooldown) / 60 })
+    : DEFAULT_LOCK_TIME
```
Comment (artsiomYavorski) [inline packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-deposit-lock-time.ts:19]: const { defaultLockTime } = useConfigContextParams()

File: packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-deposit-lock-time.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  switch (depositMethod) {
+    case 'depositWithCustomCooldown':
+    case 'nativeDepositWithCustomCooldown':
+    case 'zapDepositWithCustomCooldown':
+    case 'zapNativeDepositWithCustomCooldown':
+      return customLockTime
+    default:
+      return DEFAULT_LOCK_TIME
```
Comment (artsiomYavorski) [inline packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-deposit-lock-time.ts:28]: defaultLockTime from config

File: packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-deposit-quote.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      chainId,
+    })
+
+  const isLoading =
+    isSwapDataFetching ||
+    isDepositQuoteFetching ||
+    sendToken.value !== debouncedSendTokenValue
+
+  useEffect(() => {
+    if (!isDeposit) {
```
Comment (artsiomYavorski) [inline packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-deposit-quote.ts:38]: do we still have a use case of that hook out of deposit tab?

File: packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-deposit-slippage.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  const isDeposit = useIsDepositTradingPanelType()
+  const [, updateSettings] = useTradingPanelSettings()
+
+  const depositSlippage = useAppliedDepositSlippage()
+  const priceDiff = useDepositPriceDiff({ includesEntryFee: true })
+
+  const priceDiffDebounced = useDebounce(priceDiff - depositSlippage, 100)
+
+  useEffect(() => {
+    if (isDeposit) {
```
Comment (artsiomYavorski) [inline packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-deposit-slippage.ts:22]: do we still have a use case of that hook out of deposit tab?

File: packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-vault-deposit-transaction-arguments.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            vaultDepositTokenAddress,
+            swapData: swapData?.txData ?? '',
+            routerKey: swapData?.routerKey,
+            swapDestinationAmount: swapData?.destinationAmount ?? '0',
+            swapSlippage: depositSlippage,
+          }),
+          { value: sendTokenAmount },
+        ]
+      case 'zapDeposit':
+      case 'zapDepositWithCustomCooldown':
```
Comment (artsiomYavorski) [inline packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-vault-deposit-transaction-arguments.tsx:57]: seems like 
```
case 'zapNativeDeposit':
case 'zapNativeDepositWithCustomCooldown':
case 'zapDeposit':
case 'zapDepositWithCustomCooldown':
```
can be combined making first array's param conditional

File: packages/trading-widget/src/core-kit/const/trading.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export const EASY_SWAPPER_V2_DEPOSIT_METHODS = {
+  DEPOSIT: 'deposit',
+  DEPOSIT_CUSTOM: 'depositWithCustomCooldown',
+  NATIVE: 'nativeDeposit',
+  NATIVE_CUSTOM: 'nativeDepositWithCustomCooldown',
+  ZAP_NATIVE_DEPOSIT: 'zapNativeDeposit',
+  ZAP_NATIVE_DEPOSIT_CUSTOM: 'zapNativeDepositWithCustomCooldown',
+  ZAP_DEPOSIT: 'zapDeposit',
+  ZAP_DEPOSIT_CUSTOM: 'zapDepositWithCustomCooldown',
+} as const
```
Comment (artsiomYavorski) [inline packages/trading-widget/src/core-kit/const/trading.ts:10]: satisfies Record<string, DepositMethodName>

File: packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-deposit-quote.ts (hunk 3 above)
Comment (dimlbc) [reply packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-deposit-quote.ts:38]: it is still can be called even if withdraw tab is opened, I believe thats how @headlessui tabs work

File: packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-vault-deposit-transaction-arguments.tsx (hunk 5 above)
Comment (dimlbc) [reply packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-vault-deposit-transaction-arguments.tsx:63]: moved `buildZapDepositTransactionArguments` call into separate constant

**Review by artsiomYavorski — DISMISSED** [review]

**Review by artsiomYavorski — APPROVED** [review]

**Pull request merged**